e2e_test/batch/basic/func.slt.part

```diff
@@ -168,3 +168,13 @@ query T
 select substr(NULL, 1);
 ----
 NULL
+
+query T
+select pg_typeof('123');
+----
+unknown
+
+query T
+select pg_typeof(round(null));
+----
+Float64
```

Comment:
I doubt that if we should follow the type naming of pg. For example, Float64 should be "double precision" 
```
postgres=# select pg_typeof(round(null));
    pg_typeof
------------------
 double precision
(1 row)
```
Though it LGTM now.

Reply:
We haven been using our internal type names everywhere and never used standard sql names, so I initially avoided sql names here to keep consistency.

However, we may switch to standard sql names some day and stop confusing users with internal names. I think this may be a good place to start introducing standard names.

I also added a test case of 2-dimensional array to show that we are still not 100% same as PostgreSQL - the PostgreSQL behavior is undesirable based on my current understanding of it.